Update DeathKnightUnholy.simc

darkmullac opened the pull request:
Updated Unholy Death Knight APL logic to match SimulationCraft 11.1.5 changes:

- aoe_burst: epidemic threshold (>=), festering logic cleanup, reordered sudden_doom
- cds_aoe_san: added outbreak condition for Superstrain + 2pc (Frost Fever)
- cds_san: added second outbreak line for Frost Fever refresh
- san_st: added festering_scythe usage, extended festering_strike logic, added final wound_spender fallback

Note: This update retains `.react` expressions from the original SimC APL. I wasn’t aware of the recommendation to translate these to `.up`/`.down` until after preparing this.

darkmullac committed:
Update DeathKnightUnholy.simc

Hekili commented:
I didn't notice the `.react` expressions at first glance, but those will need to be updated.  It's just a safety precaution because there can be subtle issues with interpreting the priority if left alone.

`.react` used alone (or with `!` for negation) just needs to be changed to `.up`.
`.react` used with a numeric comparison (`.react>1`) needs to be changed to `.stack>1`.

Has this been tested pretty extensively with different talents?  Unholy APL is pretty sensitive to logic changes that impact timing; I really don't want to cause a problem that adds to the volume of tickets that are submitted here.

Hekili commented:
Oh, also, this needs to be imported in-game and the pack string (and date) in DeathKnightUnholy.lua must be updated with the export string.

darkmullac committed:
DeathKnightUnholy.simc

Fix: Replaced .react usage in Unholy DK APL for Hekili compatibility

Line 129 - Changed `buff.sudden_doom.react` to `buff.sudden_doom.up` in cds_cleave_san section

--------------------------------------------------------

Line 145 - Changed `buff.festering_scythe.react` to `buff.festering_scythe.up` in cleave section

darkmullac committed:
Update DeathKnightUnholy.simc

darkmullac committed:
Update DeathKnightUnholy.lua

darkmullac commented:
> Has this been tested pretty extensively with different talents? Unholy APL is pretty sensitive to logic changes that impact timing; I really don't want to cause a problem that adds to the volume of tickets that are submitted here.

Currently I've tested the top two most popular builds on Archon for both raid and m+. I've tried dungeons, raids and devles so far and so far it seems to be working well with epidemic being recommended correctly.

I've made the two changes you've mentioned, I believe I found all the .reacts that needed to be changed and I also believe I've done the .Lua correctly but a check would probably be best

darkmullac commented:
But I will also test various other builds over the next day or two to see if any issues arise

darkmullac commented:
After playing with the API for around a week or so now I can't see any errors or problems that jump up to me, it seems to recommended what the guide and API both say but it'd probably be ideal to have someone who plays unholy more extensively to review it to see if they can find any problems that I haven't been able to find. Maybe a user in the discord? or just someone with more expertise.

DrWren commented:
I have tried it :-) hehehe. It seems to work decently. I did have a recommendation error this weekend but the game crashed not sure what it was but hasn't happened since. 

However while raiding and watching Bicepspumps raid guide, he did mention that in the opener we fester strike first before we put down dnd? But that seems to be our first spell. I wish there was a good way to see DTs window better for our burst window or whatever. But other than that seems to be good. 

I would be interested in what others think. I pretty much play it straight as is with my cd button active and inactive no other Changes.

darkmullac commented:
> I have tried it :-) hehehe. It seems to work decently. I did have a recommendation error this weekend but the game crashed not sure what it was but hasn't happened since.
> 
> However while raiding and watching Bicepspumps raid guide, he did mention that in the opener we fester strike first before we put down dnd? But that seems to be our first spell. I wish there was a good way to see DTs window better for our burst window or whatever. But other than that seems to be good.
> 
> I would be interested in what others think. I pretty much play it straight as is with my cd button active and inactive no other Changes.

I checked the wowhead guide and the apl and they both seem to recommend dnd and then festering strike, but happy for more people to give it a test and gather some more opinions, might have to take another look and see if theres anything further that can be looked into

DrWren commented:
Here is another auto snap for you to look at:

https://pastebin.com/Hp9Pzwux

happened when I went in to a pack of mobs.

syrifgit commented:
Going to close this given the imminent 11.2 patch which will have new APLs anyways
